stable_baselines/common/schedules.py

```diff
@@ -106,3 +106,149 @@ def __init__(self, schedule_timesteps, final_p, initial_p=1.0):
     def value(self, step):
         fraction = min(float(step) / self.schedule_timesteps, 1.0)
         return self.initial_p + fraction * (self.final_p - self.initial_p)
+
+
+def get_schedule_fn(value_schedule):
+    """
+    Transform (if needed) learning rate and clip range
+    to callable.
+
+    :param value_schedule: (callable or float)
+    :return: (function)
+    """
+    # If the passed schedule is a float
+    # create a constant function
+    if isinstance(value_schedule, (float, int)):
+        # Cast to float to avoid errors
+        value_schedule = constfn(float(value_schedule))
+    else:
+        assert callable(value_schedule)
+    return value_schedule
+
+
+def constfn(val):
+    """
+    Create a function that returns a constant
+    It is useful for learning rate schedule (to avoid code duplication)
+
+    :param val: (float)
+    :return: (function)
+    """
+
+    def func(_):
+        return val
+
+    return func
+
+
+# ================================================================
+# Legacy scheduler used by A2C, AKCTR and ACER
+# ================================================================
+
+def constant(_):
+    """
+    Returns a constant value for the Scheduler
+
+    :param _: ignored
+    :return: (float) 1
+    """
+    return 1.
+
+
+def linear_schedule(progress):
+    """
+    Returns a linear value for the Scheduler
+
+    :param progress: (float) Current progress status (in [0, 1])
+    :return: (float) 1 - progress
+    """
+    return 1 - progress
+
+
+def middle_drop(progress):
+    """
+    Returns a linear value with a drop near the middle to a constant value for the Scheduler
+
+    :param progress: (float) Current progress status (in [0, 1])
+    :return: (float) 1 - progress if (1 - progress) >= 0.75 else 0.075
+    """
+    eps = 0.75
+    if 1 - progress < eps:
+        return eps * 0.1
+    return 1 - progress
+
+
+def double_linear_con(progress):
+    """
+    Returns a linear value (x2) with a flattened tail for the Scheduler
+
+    :param progress: (float) Current progress status (in [0, 1])
+    :return: (float) 1 - progress*2 if (1 - progress*2) >= 0.125 else 0.125
+    """
+    progress *= 2
+    eps = 0.125
+    if 1 - progress < eps:
+        return eps
+    return 1 - progress
+
+
+def double_middle_drop(progress):
+    """
+    Returns a linear value with two drops near the middle to a constant value for the Scheduler
+
+    :param progress: (float) Current progress status (in [0, 1])
+    :return: (float) if 0.75 <= 1 - p: 1 - p, if 0.25 <= 1 - p < 0.75: 0.75, if 1 - p < 0.25: 0.125
+    """
+    eps1 = 0.75
+    eps2 = 0.25
+    if 1 - progress < eps1:
+        if 1 - progress < eps2:
+            return eps2 * 0.5
+        return eps1 * 0.1
+    return 1 - progress
+
+
+SCHEDULES = {
```

Comment:
I think we should also take advantage of the migration to unify the scheduling, I would go for something like in PPO2/SAC/TD3, where you can pass a callable.

Reply:
Got it. I will unify these to use the newer version used by PPO2/SAC/TD3, and try to retain some of the existing schedules.

Reply:
After refreshing my head on this one, I think a mix of the classical scheduler where you get `progress` in [0, 1] and return learning-rates and such is better. The PPO2/SAC/TD3 one requires users to play around with the number of steps, which is not always clear what one "step" is, depending on the algorithm.

I suggest we skip this part for now and focus on finishing rest of the refactoring (Yesyes, me lazy).

Reply:
ok ;)